### DISCUSSION TITLE
Equivalent to `array[@]` bash array substring expansion

### DISCUSSION THREAD

**Discussion by dpetka2001** [fish-shell/fish-shell#9779] [Q&A] [answered]
As the title says is there something equivalent to `list[@]` which would denote all the elements of the list, or can one only achieve this by iterating the list with a `for` loop?

**Comment by faho** [comment, marked as answer]
`$list` by default expands to all elements of the list. If you want specific elements of the list, you can use `$list[1]` for the first, or `$list[1..3]` for the first, second and third, or `$list[-1]` for the last.

Read https://fishshell.com/docs/current/fish_for_bash_users.html#variables and https://fishshell.com/docs/current/language.html#lists for more.

**Reply by dpetka2001** [reply]
I understand. Thank you very much for your help. Will look more closely into the docs you mentioned.

**Comment by Xiretza** [comment]
Isn't that just `$list`?

```
$ set list a b c
$ printf '%s\n' $list
a
b
c
```

**Reply by dpetka2001** [reply]
Much appreciated!
